tabled/src/settings/split/mod.rs

```diff
@@ -0,0 +1,174 @@
+        Split {
+            direction: Direction::Row,
+            behavior: Behavior::Zip,
+            index,
+        }
+    }
+
+    pub fn set_behavior(self, behavior: Behavior) -> Self {
+        Self { behavior, ..self }
+    }
+}
+
+impl<RR, D, Cfg> TableOption<RR, D, Cfg> for Split
+where
+    RR: Records + ExactRecords + Resizable,
+{
+    fn change(&mut self, records: &mut RR, _: &mut Cfg, _: &mut D) {
+        let Split {
+            direction,
+            behavior,
+            index,
+        } = &self;
+        let before_rows = records.count_rows();
+        let before_columns = records.count_columns();
+
+        match (direction, behavior) {
+            (Direction::Column, Behavior::Append) => {
+                let sections_per_row =
+                    before_columns / index + { usize::from(before_columns % index != 0) };
+
+                for section in 1..sections_per_row {
+                    for reference_row in 0..before_rows {
+                        let target_row = reference_row + section * before_rows;
+                        records.push_row();
+
+                        for target_column in 0..*index {
+                            let reference_column = target_column + section * index;
+
+                            if reference_column < before_columns {
+                                records.swap(
+                                    (target_row, target_column),
+                                    (reference_row, reference_column),
+                                );
+                            }
+                        }
+                    }
+                }
+
+                for column in (*index..before_columns).rev() {
+                    records.remove_column(column)
+                }
+            }
+            (Direction::Column, Behavior::Zip) => {
+                let sections_per_row =
+                    before_columns / index + { usize::from(before_columns % index != 0) };
+
+                for reference_row in 0..before_rows {
+                    let reference_row = reference_row * sections_per_row;
+
+                    for section in 1..sections_per_row {
+                        let target_row = reference_row + section;
+
+                        records.insert_row(target_row);
+
+                        for target_column in 0..*index {
+                            let reference_column = target_column + (section * index);
+
+                            if reference_column < before_columns {
+                                records.swap(
+                                    (target_row, target_column),
+                                    (reference_row, reference_column),
+                                );
+                            }
+                        }
+                    }
+                }
+
+                for column in (*index..before_columns).rev() {
+                    records.remove_column(column)
+                }
```

Comment:
(just a note) You could create a function which would take `impl Iterator<item=usize>` and make the removal (it'd remove duplication in the branch above)

Reply:
https://github.com/zhiburt/tabled/pull/299/files#diff-881689aca954c8702c2bde0c3bf79fb1240972e785c9e672358dcc1407c05e33R204-R212